### PR TITLE
Improved docstrings in lib/_augment.pyx

### PR DESCRIPTION
**Pull request by zemanj**
Fixes (partially) #2046, this is the second of a series of related PRs, following PR #2048.

Changes made in this Pull Request:
 - Improved documentation of `MDAnalysis.lib.distances.augment_coordinates()` and `MDAnalysis.lib.distances.undo_augment()` to better explain what the methods are actually doing.
- Added corresponding `.. autofunction` in `lib/distances.py` so that the signatures of both functions are now properly displayed in the html docs.
- Corrected a mistake from PR #2048 so that `distance_array()` now also returns an array for single coordinate input.
- Added missing CHANGELOG entries for PR #2048.

Apparently, there is a problem with Windows/Linux (i.e., CRLF vs. LF) line endings resulting in an incorrect `git diff` output for `lib/_augment.pyx`, which now looks as if I changed every line in the file, even though I only changed the docstrings. With `git diff -b` the changes are displayed correctly, though. On github, just change the diff settings to ignore whitespace.

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by zemanj** [comment]
@ayushsuhane Since you wrote the augment routines (PR #1977), could you take a quick look at the new docstrings? Just to make sure I didn't add any bogus. :wink:

File: package/CHANGELOG [hunk 1]
```diff
@@ -80,7 +80,8 @@ Enhancements
   * Added dihedrals.py with Dihedral, Ramachandran, and Janin classes to
     analysis module (PR #1997, PR #2033)
   * Added the analysis.data module for reference data used in analysis (PR #2033)
-  * Added dihedrals.py with Ramachandran class to analysis module
```
Comment (richardjgowers) [inline package/CHANGELOG]: @zemanj I think you accidentally killed this entry

Comment (zemanj) [reply package/CHANGELOG]: I thought that's already included three lines above... am I wrong?

Comment (richardjgowers) [reply package/CHANGELOG]: ah it's weird it's in the log twice then, yeah ok

**Comment by zemanj** [comment]
Mmmh... nobody up to reading some good ol' poetry? :smile:

**Comment by ayushsuhane** [comment]
@zemanj  Looks good. The docs are more clear now. Thanks!! 👍

**Pull request merged**